### PR TITLE
Add new hook OnPlayerQueueRandomDungeon() and revert (now unnecessary) PR

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
## Changes Proposed:
-  Reverts https://github.com/azerothcore/azerothcore-wotlk/commit/1c43e6ac6e1a5c72c4559bd8bc9a382275d35b14 in favour of https://github.com/azerothcore/mod-rdf-expansion
-  Add new hook `OnPlayerQueueRandomDungeon()`

## Tests Performed:

- Tested in game

## How to Test the Changes:

- See https://github.com/azerothcore/mod-rdf-expansion

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Scripting/ScriptMgr.h [hunk 1]
```diff
@@ -841,6 +841,9 @@ class PlayerScript : public ScriptObject
     // Called when a player is added to battleground
     virtual void OnAddToBattleground(Player* /*player*/, Battleground* /*bg*/) { }
 
+    // Called when a player queues a Random Dungeon using the RDF (Random Dungeon Finder)
```
Comment (Winfidonarleyan) [inline src/server/game/Scripting/ScriptMgr.h:844]: Need new style info

**Comment by nolt** [comment]
Well I tested this mod and it not works as it should nor I do something wrong.
1. v3.0.0-2281-g6c953e933
2. I've made changes in ScriptMgr.h
3. run cmake, make, make install
4. While in game lvl 70 character I changed option in .conf file from 2 to 0 with .reload conf and with world restart (with game client restart) and option was always the same as on scree
Zrzut ekranu z 2021-10-01 22-00-52
n.

**Comment by locus313** [comment]
@nolt not sure if you noticed but in the module about the client limitation. Even though it displays the random tbc or wotlk dungeons, if you actually queue up it will only place you into the dungeons for the enabled expansion. 

I ran tests and queued up several times and looks to be working as expected. I only ever got put into a dungeon for the expansion that was enabled.

**Review by locus313 — APPROVED** [review]

File: src/server/game/Scripting/ScriptMgr.h (hunk 1 above)
Comment (Yehonal) [reply src/server/game/Scripting/ScriptMgr.h:844]: https://www.azerothcore.org/wiki/how-to-document-code

Comment (FrancescoBorzi) [reply src/server/game/Scripting/ScriptMgr.h:844]: sorry I missed this, I'll address this in another PR

**Pull request merged**